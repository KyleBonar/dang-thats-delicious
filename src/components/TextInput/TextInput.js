import React from "react";
import PropTypes from "prop-types";

const TextInput = ({
  id,
  name,
  onChange,
  required,
  type,
  value,
  placeholder
}) => (
  <div>
    <label htmlFor={name} className="text__upper text__grey">
      {name}
    </label>
    {type.toLowerCase() === "textarea" ? (
      <textarea
        id={id}
        name={name}
        cols="30"
        rows="10"
        placeholder={placeholder}
        onChange={onChange}
        value={value}
      />
    ) : (
      <input
        type={type}
        id={id}
        name={name}
        value={value}
        placeholder={placeholder}
        onChange={onChange}
        required={required}
      />
    )}
  </div>
);

TextInput.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  required: PropTypes.bool,
  type: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  placeholder: PropTypes.string
};

TextInput.defaultProps = {
  id: "",
  required: false,
  type: "text",
  value: "",
  placeholder: ""
};

export default TextInput;
