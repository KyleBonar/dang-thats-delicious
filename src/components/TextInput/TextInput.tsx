import * as React from "react";
import * as shortid from "shortid";

import styled from "../../theme/styled-components";
import Label from "../Label/Label";

const StyledDiv = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

const StyledInput = styled.input`
  width: 100%;
  max-width: 100%;
  padding: 10px;
  box-sizing: border-box;
  border: 1px solid #ababab;
  margin-top: 5px;
  margin-bottom: 15px;
`;
StyledInput.displayName = "StyledInput";

const StyledTextArea = styled.textarea`
  width: 100%;
  max-width: 100%;
  padding: 10px;
  box-sizing: border-box;
  border: 1px solid #ababab;
  margin-top: 5px;
  margin-bottom: 15px;
  min-width: 100%;
  min-height: 35px;
  max-height: 220px;
`;
StyledTextArea.displayName = "StyledTextArea";

interface TextInputProps {
  id?: string;
  name?: string;
  placeholder?: string;
  required?: boolean;
  label?: string;
  showLabel?: boolean;
  type?: "text" | "textarea";
  value?: string | number;
  className?: string;
  onChange(
    event:
      | React.ChangeEvent<HTMLInputElement>
      | React.ChangeEvent<HTMLTextAreaElement>
  ): void;
}

interface TextInputState {
  id: string;
}

class TextInput extends React.PureComponent<TextInputProps, TextInputState> {
  public static defaultProps = {
    showLabel: false,
    required: false,
    type: "text"
  };

  public componentWillMount() {
    // Either accept id from parent or generate unique id
    this.setState({ id: !this.props.id ? shortid.generate() : this.props.id });
  }

  public render() {
    return (
      <StyledDiv className={this.props.className}>
        {this.props.showLabel && this.props.label && (
          <Label htmlFor={this.state.id}>
            {this.props.label}
            {this.props.required ? "*" : ""}
          </Label>
        )}

        {this.props.type && this.props.type.toLowerCase() === "text" ? (
          <StyledInput
            type={this.props.type}
            id={this.state.id}
            name={this.props.name}
            value={this.props.value}
            placeholder={this.props.placeholder}
            onChange={this.props.onChange}
            required={this.props.required}
          />
        ) : (
          <StyledTextArea
            id={this.state.id}
            name={this.props.name}
            cols={30}
            rows={10}
            placeholder={this.props.placeholder}
            onChange={this.props.onChange}
            value={this.props.value}
            required={this.props.required}
          />
        )}
      </StyledDiv>
    );
  }
}

export default TextInput;
