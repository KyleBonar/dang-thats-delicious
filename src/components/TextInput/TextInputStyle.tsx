import styled from "styled-components";

export const StyledDiv = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const StyledInput = styled.input`
  width: 100%;
  max-width: 100%;
  padding: 10px;
  box-sizing: border-box;
  border: 1px solid #ababab;
  margin-top: 5px;
  margin-bottom: 15px;
`;
StyledInput.displayName = "StyledInput";

export const StyledTextArea = styled.textarea`
  width: 100%;
  max-width: 100%;
  padding: 10px;
  box-sizing: border-box;
  border: 1px solid #ababab;
  margin-top: 5px;
  margin-bottom: 15px;
  min-width: 100%;
  min-height: 35px;
  max-height: 220px;
`;
StyledTextArea.displayName = "StyledTextArea";
