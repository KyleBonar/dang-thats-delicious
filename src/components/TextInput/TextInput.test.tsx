import * as React from "react";
import * as enzyme from "enzyme";
import TextInput from "./TextInput";

describe("<TextInput />", () => {
  const onChange = jest.fn();
  const wrapper = enzyme.shallow(<TextInput onChange={onChange} />);

  it("renders", () => {
    expect(wrapper).toBeTruthy();
  });

  it("defaults to rendering single input", () => {
    expect(wrapper.find("StyledInput").length).toEqual(1);
  });
});
