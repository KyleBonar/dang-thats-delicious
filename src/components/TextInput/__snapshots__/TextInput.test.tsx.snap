// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> matches snapshot 1`] = `
<div
  className="omnis adipisci eius laborum eius aut illo"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="4ec71492-28a8-4e9b-abb8-1a04916e3258"
    name="fugiat error molestiae laborum est possimus assumenda"
    onChange={[MockFunction]}
    placeholder="enim iure facilis ut explicabo iure vitae"
    required={false}
    type="text"
    value="quia autem et modi tenetur amet modi"
  />
  <p>
    id sit delectus vel ipsa tempora aut
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 2`] = `
<div
  className="ducimus nemo doloremque maiores esse atque et"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="17ccd928-60ac-457a-8763-4633e7763072"
    onChange={[MockFunction]}
    required={false}
    type="text"
    value={[Function]}
  />
</div>
`;

exports[`<TextInput /> matches snapshot 3`] = `
<div>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="eed0deab-c662-4bfc-a685-5fda88403f33"
    name="nihil quam nulla est unde sit ex"
    onChange={[MockFunction]}
    placeholder="est doloremque explicabo eaque necessitatibus sit suscipit"
    required={false}
    type="text"
    value="et doloremque voluptate in sunt occaecati nulla"
  />
</div>
`;

exports[`<TextInput /> matches snapshot 4`] = `
<div>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="iusto sapiente nostrum aut corrupti sed consectetur"
    onChange={[MockFunction]}
    required={false}
    type="password"
    value={[Function]}
  />
  <p>
    laborum quos eum voluptate sit saepe assumenda
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 5`] = `
<div
  className="quo sunt possimus earum animi quis expedita"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="consequatur itaque et in fugit in rerum"
    name="odit aut labore qui aut inventore aut"
    onChange={[MockFunction]}
    placeholder="beatae incidunt ut ducimus fugiat et et"
    required={false}
    type="text"
    value={[Function]}
  />
</div>
`;

exports[`<TextInput /> matches snapshot 6`] = `
<div
  className="aliquam molestiae occaecati a culpa facilis eligendi"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="d3ea72b2-3925-48c4-b7d8-6fcadeb7d23c"
    name="quas repellendus in dolores totam animi ipsum"
    onChange={[MockFunction]}
    required={false}
    type="password"
    value={[Function]}
  />
  <p>
    in eum ut dignissimos aspernatur voluptate voluptatem
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 7`] = `
<div
  className="aliquid ea animi architecto accusantium vel enim"
>
  <input
    aria-disabled={true}
    aria-required={false}
    disabled={true}
    id="eius dolorem amet quia quia aut sint"
    name="sunt mollitia quibusdam officia eum repudiandae minima"
    onChange={[MockFunction]}
    required={false}
    type="text"
    value="voluptatem quia dolores dolor voluptates sequi perferendis"
  />
</div>
`;

exports[`<TextInput /> matches snapshot 8`] = `
<div
  className="occaecati eveniet aut eligendi iusto sed pariatur"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="eeb3a11d-e60c-49d1-b962-0cbad2e3bf8c"
    onChange={[MockFunction]}
    required={false}
    type="text"
    value="harum a aut esse unde ea voluptatem"
  />
  <p>
    saepe iure quia culpa suscipit totam libero
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 9`] = `
<div>
  <input
    aria-disabled={false}
    aria-required={true}
    disabled={false}
    id="quis ratione eveniet ea totam aut aut"
    name="enim rem qui hic dolore quibusdam quasi"
    onChange={[MockFunction]}
    placeholder="ut libero explicabo at quae autem eligendi"
    required={true}
    type="text"
    value="odio architecto ut quasi quasi accusantium modi"
  />
</div>
`;

exports[`<TextInput /> matches snapshot 10`] = `
<div
  className="ut illo sit rerum accusamus molestiae quia"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="in dolorem repudiandae possimus dignissimos esse qui"
    onChange={[MockFunction]}
    placeholder="molestiae quas omnis qui illo excepturi esse"
    required={false}
    type="password"
    value={[Function]}
  />
  <p>
    perspiciatis quas et quos esse dolores vitae
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 11`] = `
<div
  className="officiis omnis rem officiis et placeat ipsum"
>
  <Label
    htmlFor="molestias ut distinctio molestias placeat exercitationem unde"
  >
    voluptatem quod tempore temporibus impedit dolores saepe
  </Label>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="molestias ut distinctio molestias placeat exercitationem unde"
    onChange={[MockFunction]}
    required={false}
    type="text"
    value="voluptatibus consequatur provident qui delectus qui qui"
  />
  <p>
    id possimus quas dolor error eum incidunt
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 12`] = `
<div>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="d95caf2a-2bf5-4e08-83ce-aa2decf180ca"
    onChange={[MockFunction]}
    required={false}
    type="password"
    value="sunt libero et et ratione qui voluptate"
  />
  <p>
    delectus aliquid et dicta ipsam qui provident
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 13`] = `
<div
  className="nobis possimus eos quia quam sit optio"
>
  <Label
    htmlFor="necessitatibus blanditiis deserunt accusantium debitis aut laborum"
  >
    molestiae dolores voluptas placeat vitae ut facere
  </Label>
  <input
    aria-disabled={true}
    aria-required={false}
    disabled={true}
    id="necessitatibus blanditiis deserunt accusantium debitis aut laborum"
    name="non nulla sit officia doloremque non sit"
    onChange={[MockFunction]}
    placeholder="perferendis soluta dignissimos cum non ea sed"
    required={false}
    type="password"
    value="ea ab doloremque est nemo iste omnis"
  />
  <p>
    modi soluta hic omnis atque inventore vitae
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 14`] = `
<div>
  <Label
    htmlFor="eligendi quis excepturi veritatis occaecati odit in"
  >
    facilis est laboriosam tempora aut cupiditate aut
  </Label>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="eligendi quis excepturi veritatis occaecati odit in"
    onChange={[MockFunction]}
    placeholder="error sed asperiores amet et quos repellat"
    required={false}
    type="text"
    value={[Function]}
  />
  <p>
    minus animi quis unde animi aut sapiente
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 15`] = `
<div>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="76d0c46c-acfe-4796-a668-332aa8004049"
    onChange={[MockFunction]}
    required={false}
    type="password"
    value={[Function]}
  />
</div>
`;

exports[`<TextInput /> matches snapshot 16`] = `
<div
  className="ut eum doloribus et perspiciatis sunt eligendi"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="vel earum quo fugiat et ut voluptate"
    onChange={[MockFunction]}
    placeholder="omnis blanditiis accusamus ut odit quo enim"
    required={false}
    type="text"
    value="consequuntur et provident repellat et qui veniam"
  />
  <p>
    sit eos maxime at doloribus aut maiores
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 17`] = `
<div>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="ac1f11d4-0cb8-4420-8a11-955f7938547c"
    onChange={[MockFunction]}
    placeholder="aperiam quia est et suscipit ipsa et"
    required={false}
    type="text"
    value="voluptatem et autem doloremque earum cum et"
  />
  <p>
    autem et recusandae voluptatem possimus voluptas eaque
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 18`] = `
<div
  className="assumenda nihil ad ut maxime aliquid sint"
>
  <input
    aria-disabled={false}
    aria-required={true}
    disabled={false}
    id="303fbd6b-b3d3-45a4-a04c-39d20b0b3ad9"
    name="veritatis voluptates suscipit reiciendis in exercitationem et"
    onChange={[MockFunction]}
    required={true}
    type="text"
    value={[Function]}
  />
</div>
`;

exports[`<TextInput /> matches snapshot 19`] = `
<div
  className="nihil sint occaecati dolore autem iste nesciunt"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="1f0734d4-f59a-4dbe-b8ed-cb19ab7b7e6d"
    name="voluptatem dolores voluptas quidem excepturi temporibus facere"
    onChange={[MockFunction]}
    placeholder="velit odit ratione saepe id officia pariatur"
    required={false}
    type="password"
    value="excepturi fugit beatae ut ea rerum nobis"
  />
</div>
`;

exports[`<TextInput /> matches snapshot 20`] = `
<div
  className="quas consequuntur dolores nihil eligendi sunt aliquid"
>
  <input
    aria-disabled={true}
    aria-required={false}
    disabled={true}
    id="c76bb185-c69e-47e7-a883-a10f30b0d68d"
    onChange={[MockFunction]}
    placeholder="sed voluptas ullam animi excepturi laborum veniam"
    required={false}
    type="password"
    value={[Function]}
  />
  <p>
    dolorem porro voluptates ut amet ea repellendus
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 21`] = `
<div
  className="ab laborum sint qui id ipsa est"
>
  <Label
    htmlFor="est quis est veniam eum laudantium accusamus"
  >
    sequi magnam doloremque laudantium dolorum magni molestias
  </Label>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="est quis est veniam eum laudantium accusamus"
    onChange={[MockFunction]}
    required={false}
    type="text"
    value={[Function]}
  />
  <p>
    laudantium hic quo corporis minima cupiditate dolor
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 22`] = `
<div
  className="voluptatem iusto tempore voluptas tempore accusamus vitae"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="officiis voluptatem veniam aut voluptas suscipit sit"
    onChange={[MockFunction]}
    placeholder="nulla officia rerum incidunt porro amet at"
    required={false}
    type="password"
    value="dolorum tempore animi vel possimus eos et"
  />
  <p>
    quia unde ex magni ullam ullam doloremque
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 23`] = `
<div
  className="impedit sint similique quos rem nihil voluptatem"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="70981224-f390-42fa-b7a0-595afa5e6186"
    name="voluptatem laboriosam sed est et sapiente ipsa"
    onChange={[MockFunction]}
    placeholder="nostrum repudiandae harum neque ut dolores qui"
    required={false}
    type="text"
    value={[Function]}
  />
  <p>
    qui cum quos eos iste aut inventore
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 24`] = `
<div>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="3feba585-c472-49b3-9312-cb05d7b3616f"
    onChange={[MockFunction]}
    placeholder="iste ut a ut exercitationem voluptatem nam"
    required={false}
    type="password"
    value="doloribus et fuga adipisci saepe quam quia"
  />
</div>
`;

exports[`<TextInput /> matches snapshot 25`] = `
<div
  className="facere labore et ut unde consequatur sequi"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="d0bb27e8-0550-466a-b4cf-a06b52042efa"
    name="autem voluptas delectus iure veniam est consequatur"
    onChange={[MockFunction]}
    required={false}
    type="email"
    value="accusamus et unde quia incidunt ex et"
  />
  <p>
    vel nisi culpa provident voluptatem sit id
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 26`] = `
<div>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="tenetur iure quos sed quibusdam voluptatem ducimus"
    onChange={[MockFunction]}
    placeholder="molestiae rem sunt aut quam provident voluptates"
    required={false}
    type="password"
    value="nostrum quae et consequatur dolor doloribus cum"
  />
  <p>
    unde ut beatae similique totam quia eum
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 27`] = `
<div>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="c708ea25-9544-4959-b94b-c67732788bfb"
    onChange={[MockFunction]}
    required={false}
    type="text"
    value={[Function]}
  />
  <p>
    quos nam labore aliquid non a quod
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 28`] = `
<div>
  <input
    aria-disabled={true}
    aria-required={false}
    disabled={true}
    id="suscipit impedit veniam perferendis est aliquam consequuntur"
    name="velit quibusdam aut in blanditiis quia aliquam"
    onChange={[MockFunction]}
    placeholder="exercitationem excepturi eligendi facilis similique optio hic"
    required={false}
    type="email"
    value={[Function]}
  />
  <p>
    laboriosam occaecati expedita quam ex et nobis
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 29`] = `
<div
  className="commodi harum et reiciendis mollitia quis dolor"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="90c654ca-2f11-4f3f-b47d-32b9f98275c9"
    name="velit perspiciatis mollitia ut corrupti sunt temporibus"
    onChange={[MockFunction]}
    placeholder="consequatur porro excepturi aperiam qui voluptatem rerum"
    required={false}
    type="password"
    value="et dolorem possimus quasi accusantium quaerat voluptatem"
  />
</div>
`;

exports[`<TextInput /> matches snapshot 30`] = `
<div
  className="incidunt provident assumenda sit pariatur quam et"
>
  <input
    aria-disabled={false}
    aria-required={false}
    disabled={false}
    id="c525ab37-39ac-4ebe-a8cc-8b93c9ca1c4a"
    onChange={[MockFunction]}
    placeholder="itaque vero repellendus ex eligendi possimus animi"
    required={false}
    type="email"
    value="modi expedita quaerat eos aut itaque ut"
  />
  <p>
    aut iste explicabo expedita eos ut id
  </p>
</div>
`;

exports[`<TextInput /> matches snapshot 31`] = `
<div>
  <input
    aria-disabled={true}
    aria-required={true}
    disabled={true}
    id="est doloremque et molestiae velit sequi sapiente"
    onChange={[MockFunction]}
    placeholder="sit porro dolor corrupti sit veniam est"
    required={true}
    type="password"
    value="laborum ut non fugit sint est dolores"
  />
</div>
`;

exports[`<TextInput /> matches snapshot 32`] = `
<div
  className="possimus voluptatem nemo laboriosam est explicabo velit"
>
  <input
    aria-disabled={true}
    aria-required={false}
    disabled={true}
    id="281e5a68-de5e-4099-934a-f1d1f6ff058a"
    onChange={[MockFunction]}
    placeholder="exercitationem vitae magni a corrupti eveniet non"
    required={false}
    type="password"
    value="vel saepe et qui aut enim culpa"
  />
</div>
`;
